### PR TITLE
Bisq Network Monitor: Babysteps

### PR DESCRIPTION
**Pull request by freimair**
**Babysteps** release of the Bisq Network Monitor proposed [here](https://github.com/bisq-network/proposals/issues/62)

### CODE REVIEW

File: gradle/witness/gradle-witness.gradle [hunk 1]
```diff
@@ -10,72 +10,3 @@
...
-        'org.apache.commons:commons-compress:5f2df1e467825e4cac5996d44890c4201c000b43c0b23cffc0782d28a0beb9b0',
-        'org.tukaani:xz:a594643d73cc01928cf6ca5ce100e094ea9d73af760a5d4fb6b75fa673ecec96',
-        'com.madgag.spongycastle:core:8d6240b974b0aca4d3da9c7dd44d42339d8a374358aca5fc98e50a995764511f',
-        'net.jcip:jcip-annotations:be5805392060c71474bf6c9a67a099471274d30b83eef84bfc4e0889a4f1dcc0',
-        'org.bitcoinj:orchid:f836325cfa0466a011cb755c9b0fee6368487a2352eb45f4306ad9e4c18de080',
-        'com.squareup.okhttp:okhttp:b4c943138fcef2bcc9d2006b2250c4aabbedeafc5947ed7c0af7fd103ceb2707',
-        'org.jetbrains.kotlin:kotlin-stdlib-common:4b161ef619eee0d1a49b1c4f0c4a8e46f4e342573efd8e0106a765f47475fe39',
-        'com.squareup.okio:okio:114bdc1f47338a68bcbc95abf2f5cdc72beeec91812f2fcd7b521c1937876266',
-    ]
-}
```
Comment (cbeams) [inline gradle/witness/gradle-witness.gradle]: Was this removal intentional?

File: monitor/src/main/java/bisq/monitor/Configurable.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+package bisq.monitor;
+
+import java.util.Properties;
+
+/**
+ * Configurable is configurable.
+ * 
+ * @author Florian Reimair
+ */
```
Comment (cbeams) [inline monitor/src/main/java/bisq/monitor/Configurable.java:26]: I'd drop any boilerplate Javadoc like this and those that follow. Fine to have it if it really explains something, but otherwise it's line noise.

File: monitor/src/main/java/bisq/monitor/Configurable.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+public abstract class Configurable {
+
+
```
Comment (cbeams) [inline monitor/src/main/java/bisq/monitor/Configurable.java:30]: No need for double blank lines here

File: monitor/src/test/java/bisq/monitor/MonitorInfrastructureTests.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.monitor;
+
+
```
Comment (cbeams) [inline monitor/src/test/java/bisq/monitor/MonitorInfrastructureTests.java:21]: No need for double blank lines here

File: monitor/src/test/java/bisq/monitor/TorHiddenServiceStartupTimeTests.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+public class TorHiddenServiceStartupTimeTests {
+
+    private class DummyReporter extends Reporter {
+
+        private long result;
+
+        @Override
+        public void report(long value) {
+            result = value;
+
```
Comment (cbeams) [inline monitor/src/test/java/bisq/monitor/TorHiddenServiceStartupTimeTests.java]:
```suggestion
```

File: monitor/src/test/java/bisq/monitor/TorHiddenServiceStartupTimeTests.java [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        }
+
+        @Override
+        public void report(long value, String prefix) {
+            report(value);
+        }
+
+    }
+
+    private final static File torWorkingDirectory = new File(TorHiddenServiceStartupTimeTests.class.getSimpleName());
```
Comment (cbeams) [inline monitor/src/test/java/bisq/monitor/TorHiddenServiceStartupTimeTests.java]: Please declare fields together at top of class, not interspersed throughout methods.

File: monitor/src/test/java/bisq/monitor/TorStartupTimeTests.java [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.monitor;
+
+
```
Comment (cbeams) [inline monitor/src/test/java/bisq/monitor/TorStartupTimeTests.java:21]: No need for double blank lines here

File: build.gradle [hunk 8]
```diff
@@ -311,14 +311,34 @@ configure(project(':desktop')) {
...
+        }
+    }
 
     dependencies {
-        compile project(':core')
-        compile "com.sparkjava:spark-core:$sparkVersion"
-        compile 'net.gpedro.integrations.slack:slack-webhook:1.1.1'
+        compile 'org.slf4j:slf4j-api:1.7.22'
+        compile 'ch.qos.logback:logback-core:1.1.10'
+        compile 'ch.qos.logback:logback-classic:1.1.10'
```
Comment (cbeams) [inline build.gradle]: Since these dependencies are now repeated between `:monitor` and `:common`, their version numbers can be extracted to `$sf4jVersion` and `$logbackVersion` to avoid version drift between the two subprojects.

**Review by cbeams — CHANGES_REQUESTED** [review]
This was a very quick review from my side. Mostly style nits, but my question about the removal of gradle-witness data should be addressed before approval.

**Comment by ManfredKarrer** [comment]
Will review once back from holiday in January.

File: monitor/src/main/java/bisq/monitor/Configurable.java (hunk 2 above)
Comment (freimair) [reply monitor/src/main/java/bisq/monitor/Configurable.java:26]: agreed. Updated. I fear that I have been too much in a hurry to get the thing up and running :)

File: monitor/src/main/java/bisq/monitor/Configurable.java (hunk 3 above)
Comment (freimair) [reply monitor/src/main/java/bisq/monitor/Configurable.java:30]: done

File: monitor/src/test/java/bisq/monitor/MonitorInfrastructureTests.java (hunk 4 above)
Comment (freimair) [reply monitor/src/test/java/bisq/monitor/MonitorInfrastructureTests.java:21]: done

File: monitor/src/test/java/bisq/monitor/TorStartupTimeTests.java (hunk 7 above)
Comment (freimair) [reply monitor/src/test/java/bisq/monitor/TorStartupTimeTests.java:21]: done

File: monitor/src/test/java/bisq/monitor/TorHiddenServiceStartupTimeTests.java (hunk 6 above)
Comment (freimair) [reply monitor/src/test/java/bisq/monitor/TorHiddenServiceStartupTimeTests.java]: only the private class is declared above. Old habits. Done.

File: monitor/src/test/java/bisq/monitor/TorHiddenServiceStartupTimeTests.java (hunk 5 above)
Comment (freimair) [reply monitor/src/test/java/bisq/monitor/TorHiddenServiceStartupTimeTests.java]: done

File: build.gradle (hunk 8 above)
Comment (freimair) [reply build.gradle]: agreed. However, as the next steps for the Monitoring Node is integrating Bisq Protocol foo, I kept the dependencies separate as the :monitor project will depend on :common eventually.

However, done.

File: gradle/witness/gradle-witness.gradle (hunk 1 above)
Comment (freimair) [reply gradle/witness/gradle-witness.gradle]: ahm yes and no. As the monitor node is under heavy development, I removed the witness stuff as I had to fix Netlayer, tor-binary and jtorctl dependencies over and over again. I, of course, agree to supply a release with the proper checksums.

done.

**Comment by ManfredKarrer** [comment]
Is bisq.monitor.metric.Dummy.java still needed (not used yet)?

**Review by ManfredKarrer — APPROVED** [review]
ACK
